tools/skbstat.py

```diff
@@ -0,0 +1,1079 @@
+#!/usr/bin/python
+# @lint-avoid-python-3-compatibility-imports
+#
+# skbstat Trace skb events(input/output/consume/drop) of kernel.
+#         For Linux, uses BCC, eBPF. Embedded C.
+#
+# usage: skbstat.py [-h] [-i INTERVAL] [-s SADDR] [-d DADDR] [-N ADDR]
+#                   [-S SPORT] [-D DPORT] [-P PORT] [-p PROTO] [--input]
+#                   [--output] [--consume] [--no_drop]
+#
+# This uses dynamic tracing of kernel functions, and will need to be updated
+# to match kernel changes.
+#
+# Copyright (c) 2021 Dongdong Wang <wangdongdong.6@bytedance.com>
+# Licensed under the Apache License, Version 2.0 (the "License")
+#
+# 10-MAR-2021  Dongdong Wang Created this.
+
+from __future__ import print_function
+from bcc import BPF
+import sys
+import argparse
+import ipaddress
+from time import strftime
+from struct import unpack
+from time import sleep
+
+# arguments
+examples = """examples:
+    skbstat                               # dump kernel skb drop backtraces (default)
+    skbstat --no_drop                     # do not dump kernel skb drop backtraces
+    skbstat --input                       # dump kernel skb input backtraces
+    skbstat --output                      # dump kernel skb output backtraces
+    skbstat --consume                     # dump kernel skb consume routines
+    skbstat -s 10.0.0.1                   # only trace packet comming from 10.0.0.1
+    skbstat -N 10.0.0.1                   # only trace packet comming from or going to 10.0.0.1
+    skbstat -N 1000::1                    # only trace packet comming from or going to 1000::1
+    skbstat -p tcp                        # only trace TCP packet
+    skbstat -p tcp -D 80                  # only trace input HTTP packet
+    skbstat -p tcp -P 80                  # only trace input or output HTTP packet
+"""
+parser = argparse.ArgumentParser(
+    description="""
+    Trace skb event (input/output/consume/drop) of the kernel.
+
+    WARNING: When run with `--input/output/consume` options, 
+    this tool will trace high traffic kernel functions and may lead to 
+    some noticeable overhead. Use it carefully on production system.
```

Comment:
There is no need to put a warning description here. Just put in the beginning of the file around line 13.

Reply:
agree.